main/src/main/java/org/mobilitydata/gtfsvalidator/validator/FrequencyTimeInOrderValidator.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2020 Google LLC, MobilityData IO
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.validator;
+
+import org.mobilitydata.gtfsvalidator.annotation.GtfsValidator;
+import org.mobilitydata.gtfsvalidator.notice.NoticeContainer;
+import org.mobilitydata.gtfsvalidator.notice.StartAndEndTimeOutOfOrderNotice;
+import org.mobilitydata.gtfsvalidator.table.GtfsFrequency;
+import org.mobilitydata.gtfsvalidator.type.GtfsTime;
+
+import static org.mobilitydata.gtfsvalidator.table.GtfsFrequencyTableLoader.FILENAME;
+
+/**
+ * Validates `frequencies.start_time` is before or equal to `frequencies.end_time` for a single entity.
+ * <p>
+ * Generated notices:
+ * * StartAndEndTimeOutOfOrder
+ */
+@GtfsValidator
+public class FrequencyTimeInOrderValidator extends SingleEntityValidator<GtfsFrequency> {
+    @Override
+    public void validate(GtfsFrequency frequency, NoticeContainer noticeContainer) {
+        // validate() will only be called if startTime and endTime have been populated for this frequency
+        GtfsTime startTime = frequency.startTime();
+        GtfsTime endTime = frequency.endTime();
+        if (startTime.isAfter(endTime)) {
```

Comment:
We at Google require `start_time < end_time`, so they may not be equal. I do not see a use case for `start_time == end_time` (covered by a unit test here) - providers should use non-frequency-based trips in this case.

Reply:
Thanks @aababilov, that's interesting to know. That's what I originally thought, but IIRC others said that having these values as equal is a potentially valid case if you want a vehicle to circulate once. What you say makes sense to me, but we need to update the spec to reflect this.

@MobilityData/transit-specs another spec clarification issue.

Reply:
IMHO, if `start_time == end_time`, it is definitively a wrong use of frequencies.txt, but it should not make a dataset invalid. As Sean said, it's a potentially valid case if you want a vehicle to circulate once. This should then be treated as a warning.

Registered in Discrepancies Specification vs Validator MobilityData/gtfs-tasks#87

Reply:
Currently, the validator does not generate any notice if `start_time == end_time`

Reply:
Opening an issue on this matter so that we can revisit this https://github.com/MobilityData/gtfs-validator/issues/753